src/Mvc/Mvc.Testing/src/Handlers/RedirectHandler.cs

```diff
@@ -104,7 +105,13 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
     {
         foreach (var header in originalRequestHeaders)
         {
-            newRequestHeaders.TryAddWithoutValidation(header.Key, header.Value);
+            // Avoid copying the Authorization header to match the behavior
+            // in the built-in HTTP client when processing redirects
+            // https://github.com/dotnet/runtime/blob/69b5d67d9418d672609aa6e2c418a3d4ae00ad18/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs#L509-L517
+            if (header.Key != HeaderNames.Authorization)
```

Comment:
Does this need a `OrdinalIgnoreCase` comparison? Guessing that's a string on the LHS, no?